src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMStep.java

```diff
@@ -19,129 +19,82 @@
+        this.serverId = serverId;
+
         if (isBlank(projectName)) {
-            LOGGER.info("Error creating the Bitbucket SCM: The project name is blank");
-            projectKey = "";
-            repositorySlug = "";
-            selfLink = "";
-            cloneUrl = "";
-            repositoryId = -1;
-            return;
+            throw new BitbucketSCMException("Error creating the Bitbucket SCM: The project name is blank");
         }
+        this.projectName = projectName;
+
         if (isBlank(repositoryName)) {
-            LOGGER.info("Error creating the Bitbucket SCM: The repository name is blank");
-            projectKey = "";
-            repositorySlug = "";
-            selfLink = "";
-            cloneUrl = "";
-            repositoryId = -1;
-            return;
+            throw new BitbucketSCMException("Error creating the Bitbucket SCM: The repository name is blank");
         }
-        BitbucketRepository repository;
-        String repoCloneUrl;
-        if (!isBlank(mirrorName)) {
-            try {
-                EnrichedBitbucketMirroredRepository mirroredRepository =
-                        descriptor.createMirrorHandler(scmHelper)
-                                .fetchRepository(
-                                        new MirrorFetchRequest(
-                                                serverConfiguration.getBaseUrl(),
-                                                credentialsId,
-                                                globalCredentialsProvider,
-                                                projectName,
-                                                repositoryName,
-                                                mirrorName));
-                repository = mirroredRepository.getRepository();
-                repoCloneUrl = getCloneUrl(mirroredRepository.getMirroringDetails().getCloneUrls());
-            } catch (MirrorFetchException ex) {
-                projectKey = "";
-                repositorySlug = "";
-                selfLink = "";
-                cloneUrl = "";
-                repositoryId = -1;
-                return;
-            }
-        } else {
-            repository = scmHelper.getRepository(projectName, repositoryName);
-            repoCloneUrl = getCloneUrl(repository.getCloneUrls());
-        }
-        this.cloneUrl = repoCloneUrl;
-        projectKey = repository.getProject().getKey();
-        repositorySlug = repository.getSlug();
-        selfLink = repository.getSelfLink();
-        repositoryId = repository.getId();
+        this.repositoryName = repositoryName;
+    }
+
+    @DataBoundSetter
+    public void setId(String id) {
+        this.id = requireNonNull(id, "id");
+    }
+
+    @DataBoundSetter
+    public void setCredentialsId(@Nullable String credentialsId) {
+        this.credentialsId = StringUtils.stripToNull(credentialsId);
```

Comment:
Many of these fields are kept as null. I usually prefer blank strings, but a blank string will appear in the snippet as `credentialId: ''`. If it is null, the field is absent entirely, which I believe is preferred behaviour.